src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -0,0 +1,610 @@
+							updateStreamStats(stats, stream);
+							continue;
+						}
+
+						// Stream status filtering
+						if (this.streamStatus != null && !streamStatusSet.contains(streamStats.getState())) {
+							continue;
+						}
+
+						// Port number filtering
+						if (this.portNumber != null) {
+							Integer port = Integer.parseInt(streamInfo.getPort());
+							boolean isValidPort = handleAdapterPortRangeFromUser(port);
+							if (!isValidPort) {
+								continue;
+							}
+						}
+						updateStreamStats(stats, stream);
+					}
+				} else {
+					updateStreamStatisticsFailedMonitor(failedMonitor);
+				}
+			} else {
+				updateStreamStatisticsFailedMonitor(failedMonitor);
+			}
+		} catch (Exception e) {
+			updateStreamStatisticsFailedMonitor(failedMonitor);
+		}
+	}
+
+	/**
+	 * Counting metric group is failed to monitor
+	 *
+	 * @return number failed monitoring metric group in the metric
+	 */
+	private int getNoOfFailedMonitorMetricGroup() {
+		int noOfFailedMonitorMetric = 3;
+		noOfFailedMonitorMetric += EnumSet.allOf(MonitoringMetricGroup.class).stream().count();
+		return noOfFailedMonitorMetric;
+	}
+
+	/**
+	 * This method is used to populate all monitoring properties:
+	 * <li>Decoders statistic</li>
+	 * <li>Streams statistic</li>
+	 *
+	 * @param stats list statistic property
+	 * @throws ResourceNotReachableException when failedMonitor said all device monitoring data are failed to get
+	 */
+	private void populateDecoderMonitoringMetrics(Map<String, String> stats) {
+		Objects.requireNonNull(stats);
+
+		if (!StringUtils.isNullOrEmpty(getPassword()) && !StringUtils.isNullOrEmpty(getLogin())) {
+			retrieveSessionFromDecoder();
+			retrieveUserRoleBasedFromDecoder();
+		} else {
+			this.authenticationCookie.setSessionID(DecoderConstant.AUTHORIZED);
+		}
+		// Retrieving all device info
+		retrieveDeviceInfo(stats);
+
+		// Retrieving all decoders stats
+		for (int decoderID = DecoderConstant.MIN_DECODER_ID; decoderID < DecoderConstant.MAX_DECODER_ID; decoderID++) {
+			retrieveDecoderStats(stats, decoderID);
+		}
+
+		// Retrieving all streams stats
+		retrieveStreamStats(stats);
+
+		if (failedMonitor.size() == getNoOfFailedMonitorMetricGroup()) {
+			authenticationCookie.setSessionID(null);
+			StringBuilder errBuilder = new StringBuilder();
+			for (Map.Entry<String, String> failedMetric : failedMonitor.entrySet()) {
+				errBuilder.append(failedMetric.getValue());
+				errBuilder.append(DecoderConstant.NEXT_LINE);
+			}
+			throw new ResourceNotReachableException(errBuilder.toString());
+		}
+	}
+
+	/**
+	 * This method is used to handle  input from adapter properties and convert it to Set of String  for control
+	 *
+	 * @return Set<String> is the Set of String of filter element
+	 */
+	public Set<String> handleAdapterPropertiesInputFromUser(String input) {
+		String[] listAdapterPropertyElement = input.split(String.valueOf(DecoderConstant.COMMA));
```

Comment:
please remove String.valueOf. The DecoderConstant.COMMA is the string :)

Reply:
I fixed it